task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -195,13 +195,12 @@ def xcom_pull(
         if TYPE_CHECKING:
             assert isinstance(msg, XComResult)
 
-        value = msg.value
-        if value is not None:
+        if msg.value is not None:
             from airflow.models.xcom import XCom
 
             # TODO: Move XCom serialization & deserialization to Task SDK
             #   https://github.com/apache/airflow/issues/45231
-            return XCom.deserialize_value(value)
+            return XCom.deserialize_value(msg)  # type: ignore[arg-type]
```

Comment:
https://github.com/apache/airflow/blob/9be59712a8934c9c7eb90a478964a2e0f8260ecf/airflow/models/xcom.py#L460-L473

This method expects it to have `value` so we pass it and ignore type check for reasons mentioned in the PR description